### PR TITLE
Eventhub archive

### PR DESCRIPTION
**Pull request by ShubhaVijayasarathy**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
*
*
*
*
*

### Description of the change

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@ShubhaVijayasarathy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (shvija). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: 201-eventhubs-create-namespace-and-enable-archive/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# 201-eventhubs-create-namespace-and-enable-archive
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-event-hubs-create-event-hub-and-enable-archive%2Fazuredeploy.json" target="_blank">
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/README.md]: this link is incorrect; it should be 201-eventhubs-create-namespace-and-endable-archive

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 2]
```diff
@@ -0,0 +1,152 @@
+{  
+   "$schema":"http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: Is it possible to use a non-preview schema instead? e.g. https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            "description":"Name of the EventHub namespace"
+         }
+      },
+      "eventHubName":{  
+         "type":"string",
+         "metadata":{  
+            "description":"Name of the Event Hub"
+         }
+      },
+	  "messageRetentionInDays":{
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: formatting is off. Can be fixed later though

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+		}
+	  },
+	  
+	  "archiveTime":{
+		"type":"int",
+		"defaultValue":300,
+		"minValue":60,
+		"maxValue":900,
+		"metadata":{
+			 "description":"the time window for the archival"
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: Should specify the unit of time

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+		}
+	  },
+	  "archiveSize":{
+		"type":"int",
+		"defaultValue":314572800,
+		"minValue":10485760,
+		"maxValue":524288000,
+		"metadata":{
+			"description":"the size window for archival"
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: Should specify the unit of size

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+		}
+	  },
+	  "destinationStorageAccountResourceId":{
+		"type":"string",
+		"metadata":{
+			"description":"Storage Account resource id where you want the blobs be archived"
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: please add "existing" to the description to make it clear that the storage account needs to already exist. Also, if you could add a sentence to the README saying that the storage account needs to be pre-created that would be awesome

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+		}
+	  },
+	  "blobContainerName":{
+		"type":"string",
+		"metadata":{
+			"description":"Storage Container that you want the blobs archived in"
+		}
+	  },
+	  
+      "apiVersion":{
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: why is this a parameter? Is this something the user will actually need to change?

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+            "name":"Standard",
+            "tier":"Standard"
+         },
+         "resources":[  
+            {  
+               "apiVersion":"[variables('ehVersion')]",
+               "name":"[parameters('eventHubName')]",
+               "type":"EventHubs",
+               "dependsOn":[  
+                  "[concat('Microsoft.EventHub/namespaces/', parameters('eventHubNamespaceName'))]"
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: I don't see an eventHubNamespace actually made in this template; does this depensOn actually do anything?

File: 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+			"description":"the size window for archival"
+		}
+	  },
+	  "destinationStorageAccountResourceId":{
+		"type":"string",
+		"metadata":{
+			"description":"Storage Account resource id where you want the blobs be archived"
+		}
+	  },
+	  "blobContainerName":{
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/azuredeploy.JSON]: Does the container need to be pre-created too? Please clarify in the metadata description and the README.

**Comment by gatneil** [comment]
merging but requires post-merge fixes

File: 201-eventhubs-create-namespace-and-enable-archive/README.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# 201-eventhubs-create-namespace-and-enable-archive
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F201-eventhubs-create-namespace-and-enable-archive%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
```
Comment (gatneil) [inline 201-eventhubs-create-namespace-and-enable-archive/README.md]: Also, please add visualize button

**Pull request merged**